### DISCUSSION TITLE
how to get provider and link walletConnect v2 to a smart contract using ethers

### DISCUSSION THREAD

**Discussion by Ronexlemon** [WalletConnect/walletconnect-monorepo#2922] [Developer Support]
how to get provider and link walletConnect v2 to a smart contract using ethers

**Comment by crypblizz8** [comment]
Can you provide more context?

You would use a package like `ethereum-provider` and then use the methods provided by `ethers` to read and write to smart contracts.

Example [here](https://ethereum.stackexchange.com/questions/120817/how-to-call-a-contract-function-method-using-ethersjs).

**Reply by Ronexlemon** [reply]
here is my provider 
export const initializeProviders = async () => {
  const providers = await UniversalProvider.init({
    logger: 'info',
    // relayUrl: 'ws://<relay-url>',
    projectId: projectId,
    metadata: {
      name: 'React App',
      description: 'React App for WalletConnect',
      url: 'https://walletconnect.com/',
      icons: ['https://avatars.githubusercontent.com/u/37784886'],
    },
    client: undefined, // optional instance of @walletconnect/sign-client
  });

  // Create sub providers for each namespace/chain
  await providers.connect({
    namespaces: {
      eip155: {
        methods: [
          'eth_sendTransaction',
          'eth_signTransaction',
          'eth_sign',
          'personal_sign',
          'eth_signTypedData',
        ],
        chains: ['eip155:4002'],
        events: ['chainChanged', 'accountsChanged'],
        rpcMap: {
          4002:
            'https://rpc.walletconnect.com?chainId=eip155:4002&projectId=<your walletconnect project id>',
        },
      },
    },
   // pairingTopic: '<123...topic>', // optional topic to connect to
    skipPairing: false, // optional to skip pairing (later it can be resumed by invoking .pair())
  });

  return providers;
};

**Reply by Ronexlemon** [reply]
this how am getting calling the provider 
const getProvider = async()=>{
    const newProvider = await initializeProviders();
    const wprovider = new ethers.providers.Web3Provider(newProvider);
    setWeb3Provider(wprovider);
    console.log("provides...................",await wprovider.getNetwork());
    
    

  }

**Reply by Ronexlemon** [reply]
and this is how am calling the smart contract 
if (web3Provider) {
        
      const {chainId} = await web3Provider.getNetwork();
      
      //const code = await web3Provider.getCode(ChainLinkContract);
      const sign2 =  web3Provider.getSigner();
      
      const contract = new Contract(
        ShareFlowContract,
        shareAbi,
        web3Provider
      );
      console.log("data is loaing...........");
      const data=  await contract.functions.isCollateralTokenListed(FantomContractAddress);
      console.log("data is loaing end...........");
     // data.wait();
      console.log("data is",data);

**Reply by Ronexlemon** [reply]
but nothing is happening no wallets opens to confirm my transaction

**Comment by Ronexlemon** [comment]
Thats what  keeps me a bit confusing   any where i can get a well written
documentation   how i can use ethereum provider ... and be able to call a
smart contract

**Reply by devon-n** [reply]
I'm having the same problem. Nothing comes up on the mobile wallet
